From 6861aca5cbbe9320dde9af6eeddc1bfcc6cf642c Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 16 Mar 2017 12:15:50 +0200
Subject: [PATCH 1619/5242] MLK-14473: hdmi: Fix ioctl implementation

commit  b89e70279fa215f52827994182846ddc9c5c17a9 from
https://source.codeaurora.org/external/imx/linux-imx.git

The function mxc_hdmi_ioctl is passing kernel memory to user-space. The
case for HDMI_IOC_GET_CPU_TYPE is passing the memory directly, which is
not permitted. Fixed this, by using put_user().

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_hdmi.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mxc_hdmi.c b/drivers/video/fbdev/mxc/mxc_hdmi.c
index 513ab16..830d518 100644
--- a/drivers/video/fbdev/mxc/mxc_hdmi.c
+++ b/drivers/video/fbdev/mxc/mxc_hdmi.c
@@ -2733,13 +2733,14 @@ static long mxc_hdmi_ioctl(struct file *file,
 				sizeof(g_hdmi->hdmi_data)) ? -EFAULT : 0;
 		break;
 	case HDMI_IOC_GET_CPU_TYPE:
-		*argp = g_hdmi->cpu_type;
+		ret = put_user(g_hdmi->cpu_type, argp);
 		break;
 	default:
 		pr_debug("Unsupport cmd %d\n", cmd);
 		break;
-     }
-     return ret;
+	}
+
+	return ret;
 }
 
 static int mxc_hdmi_release(struct inode *inode, struct file *file)
-- 
1.7.9.5

